js/dragDrop.js

```diff
@@ -0,0 +1,40 @@
+$(document).ready(function () {
+    dragDropOps.dragOps();
+    dragDropOps.dropOps();
+    dragDropOps.binOps();
+});
+
+var dragDropOps = {
+    dragOps: function () {
+        $(".product").draggable({
+            helper: "clone",
+            revert: "invalid"
+        });
+    },
+
+    dropOps: function() {
+        $(".cart").droppable({
+            activeClass: "ui-state-highlight",
+            hoverClass: "ui-state-hover",
+            accept: '.product:not(".clones")',
+            drop: function (event, ui) {
+                $(this).append($(ui.draggable).clone().addClass('clones'));
+                $('.clones').draggable({
```

Comment:
In the first draggable there was a property helper clone, do we not need that here?